tests/e2e/new_mnemonic.rs

```diff
@@ -198,7 +259,64 @@ fn test_withdrawal_credentials_bls() -> Result<(), Box<dyn std::error::Error>> {
     cmd.arg("--num_validators");
     cmd.arg(num_validators);
     cmd.arg("--withdrawal_credentials");
-    cmd.arg(bls_withdrawal_credentials);
+    cmd.arg(execution_withdrawal_credentials);
+
+    cmd.assert().success();
+
+    // read generated output
+
+    let output = &cmd
+        .output()
+        .expect("could not get output from command")
+        .stdout;
+    let command_output = std::str::from_utf8(output).expect("could not parse output into string");
+    let generated_validator_json: ValidatorExports =
+        serde_json::from_str(command_output).expect("could not unmarshal command output");
+    let generated_private_key = generated_validator_json
+        .private_keys
+        .get(0)
+        .expect("could not get generated private key");
+    let generated_deposit_data = generated_validator_json
+        .deposit_data
+        .get(0)
+        .expect("could not get generated deposit data")
+        .to_owned();
+
+    generated_deposit_data.validate();
+
+    // decrypt keystore with expected password to derive private key
+    let encoded_private_key = decrypt_generated_keystore(
+        generated_validator_json.keystores.get(0).unwrap(),
+        decryption_password,
+    );
+    assert_eq!(generated_private_key, &encoded_private_key);
+
+    Ok(())
+}
+
+/*
+    generate 1 validator with new mnemonic with kdf pbkdf2 specified
+*/
+#[test]
+fn test_keystore_kdf_pbkdf2() -> Result<(), Box<dyn std::error::Error>> {
+    let chain = "goerli";
+    let decryption_password = "testtest";
+    let num_validators = "1";
+    let kdf = "pbkdf2";
+
+    // run eth-staking-smith
+
+    let mut cmd = Command::cargo_bin("eth-staking-smith")?;
+
+    cmd.arg("new-mnemonic");
+    cmd.arg("--chain");
+    cmd.arg(chain);
+    cmd.arg("--keystore_password");
+    cmd.arg(decryption_password);
+    cmd.arg("--num_validators");
+    cmd.arg(num_validators);
+    cmd.arg("--kdf");
+    cmd.arg(kdf);
 
     cmd.assert().success();
```

Comment:
This should also test that generated keystore has proper encryption specified, e.g.

given generated keystore record is like
```
{
  "crypto": {
    "checksum": {
      "function": "sha256",
      "message": "b95e2bf5c5aafe0e9149f08d2e38f1daa5d20b73972433cd1b523e647249544c"
    },
    "cipher": {
      "function": "aes-128-ctr",
      "message": "bbed72f35ea11f66a44ca42c5b593574db1fb1bd763c91e64a8629899fb58ffe",
      "params": {
        "iv": "793b3201f800b017e80c35b5486ab5df"
      }
    },
    "kdf": {
      "function": "pbkdf2",
      "message": "",
      "params": {
        "dklen": 32,
        "n": 262144,
        "p": 1,
        "r": 8,
        "salt": "14260c0c72c4afcde2b99ccdf2e1b1257d3cb5dfdc19ddbdc70f2b36eee90b9a"
      }
    }
  },
  "description": "",
  "path": "m/12381/3600/9/0/0",
  "pubkey": "b9d798bbff9a7d7599617035c1141ef607f0b4398ad9c3753219c634397b0a6f6f162b280f9322e41e4878d3d1a351df",
  "uuid": "8e32aff9-96f2-4e87-80c7-0db7ab1f8327",
  "version": 4
}
```

the test logic should parse json and assert /kdf/function property is "pbkdf2"

Reply:
added in [d939b31](https://github.com/ChorusOne/eth-staking-smith/pull/18/commits/d939b31a75957c2f2769a54fdd8ff69cadb9579d)